packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1364,6 +1403,38 @@ export class SnapController extends BaseController<
     }
   }
 
+  /**
+   * Handles increasing the openSessions counter on session open.
+   *
+   * @param snapId - The snap id of the snap the session was opened with.
+   */
+  onSessionOpen(snapId: SnapId) {
+    const runtime = this.getRuntimeOrDefault(snapId);
+    runtime.openSessions += 1;
+  }
+
+  /**
+   * Handles decreasing the openSessions counter on session close.
+   *
+   * @param snapId - The snap id of the snap the session was closed with.
+   */
+  onSessionClose(snapId: SnapId) {
+    const runtime = this.getRuntimeOrDefault(snapId);
+    runtime.openSessions -= 1;
```

Comment:
adding `assert(runtime.openSessions >= 0)` would be nice

Reply:
That should probably be `assert(runtime.openSessions > 0)`? Otherwise it could be set to `-1`.

Reply:
What? It is allowed to be `0` 🤔

Reply:
```js
let openSessions = 0;
console.assert(openSessions >= 0);
openSessions -= 1;
console.log(openSessions); // -1
```

Even if you put the assert after the decrement, the value would still be changed.

Reply:
Oh, you mean if I assert before doing the operation.